solutions/ojuz-knapsack.mdx

```diff
@@ -0,0 +1,171 @@
+---
+id: ojuz-knapsack
+source: Singapore NOI
+title: knapsack
+author: Kevin Sheng
+---
+
+## Explanation
+
+Note that the bound on $S$ is rather small and that $1 \leq W_i \leq S$, so we can group the items together by weight.  
+After this, we can implement the knapsack problem on the grouped together items. We also sort items of the same weight by reverse value, as it is always optimal to select the item with greater value among two items of the same weight.  
```

Comment:
I think you should also write an editorial for subtasks 1-4 which serves as a great introduction to multiple pack problem and the thought process of converting it to a 01 knapsack problem.

Reply:
i'd do that if i were smarter ahahahaha